Cindy/org app table rebase

cindyxzhang opened the pull request:
_Describe your feature here_
So I rebased my last PR and I followed a stackoverflow post which told me to make a new branch so it opened a new PR! I'm kinda confused on how I got here tbh, but this PR should include my fixes from the other org app table PR!
## Related PRs

_Optional - any related PRs you're waiting on, or PRs that will conflict, etc_

## Migrations

_Optional - if you added anything to the database through migration(s)_

## Screenshots

_Required for visual changes_

CC: @kwfk

cindyxzhang committed:
feat(new table): added new table

cindyxzhang committed:
feat(rebasing main for branch): rebasing main from freds comment

cindyxzhang committed:
feat(fixed table fields): fixed table fields

cindyxzhang committed:
fix(bug fixes in org table name): fixing name

kwfk commented on migrations/20201021221228-create-organization_applications.ts:
```diff
@@ -0,0 +1,72 @@
+  db.runSql(
+    `
+      CREATE TYPE application_status AS ENUM ('draft', 'submitted', 'approved', 'rejected');
+      CREATE TYPE organization_type AS ENUM ('510(c)(3)', 'Grassroots', 'Other');
```
Change to 
501(c)(3)
Grassroots/Local
Statewide
National
Other

kwfk commented on migrations/20201021221228-create-organization_applications.ts:
```diff
@@ -0,0 +1,72 @@
+    `
+      CREATE TYPE application_status AS ENUM ('draft', 'submitted', 'approved', 'rejected');
+      CREATE TYPE organization_type AS ENUM ('510(c)(3)', 'Grassroots', 'Other');
+      CREATE TYPE work_type AS ENUM ('Advocacy', 'Direct Service', 'Networking');
```
Change 'Networking' to 'Networking/Social'

kwfk commented on migrations/20201021221228-create-organization_applications.ts:
Change to type DECIMAL

kwfk commented on migrations/20201021221228-create-organization_applications.ts:
change to TEXT

kwfk commented on migrations/20201021221228-create-organization_applications.ts:
leave lowercase: ein

kwfk reviewed (changes requested):
minor changes but looks good

cindyxzhang committed:
fix(fixed fields): fixed fields according to fred edits

kwfk committed:
fix(schema.prisma): re-run db:update with updated script so that transform works correctly

Updated Prisma schema so that it is transformed before it is generated

kwfk committed:
feat(organization applications): move foreign key to Organizations in relationship with org app

Created a new migration to add a foreign key column to Organizations table. Changed
OrganizationApplications table to make most fields nullable.

kwfk reviewed (approved):
I made some changes so take a look at the commit I made:
- I moved the foreign key to the Organization table for the relationship with OrganizationApplications because organizations will be created after the application.
- I also made many of the fields nullable because most of the validation will happen in the backend and frontend rather than the db.

cindyxzhang committed:
Merge branch 'main' into cindy/org-app-table-rebase
